### PR TITLE
[Search]: Fix search modal context scope

### PR DESCRIPTION
**Pull request by camilaibs**
## Hey, I just made a Pull Request!

To allow people to use a global search context in the search modal, this pull request changes the search modal code to only create a local search context for the modal if there is no parent context already defined.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/search/src/components/SearchModal/SearchModal.tsx [hunk 1]
```diff
@@ -170,6 +170,17 @@ export const Modal = ({ toggleModal }: SearchModalProps) => {
   );
 };
 
+const Context: FC = ({ children }) => {
```
Comment (freben) [inline plugins/search/src/components/SearchModal/SearchModal.tsx:173]: avoid FC and SFC :) https://github.com/backstage/backstage/blob/master/docs/architecture-decisions/adr006-avoid-react-fc.md

Comment (freben) [reply plugins/search/src/components/SearchModal/SearchModal.tsx:173]: possible ideal: `function Context(props: { children?: React.ReactNode }) {...}` (avoids early unpacking, adds named function)

Comment (camilaibs) [reply plugins/search/src/components/SearchModal/SearchModal.tsx]: Fixed, thanks 🙌🏻

File: plugins/search/src/components/SearchModal/SearchModal.tsx [hunk 2]
```diff
@@ -170,6 +170,17 @@ export const Modal = ({ toggleModal }: SearchModalProps) => {
   );
 };
 
+const Context = ({ children }: PropsWithChildren<{}>) => {
```
Comment (emmaindal) [inline plugins/search/src/components/SearchModal/SearchModal.tsx:173]: maybe its possible to use the [`useSearchContextCheck`](https://github.com/backstage/backstage/blob/master/plugins/search-react/src/context/SearchContext.tsx#L87) hook here? 

Option 1:

```tsx
const Context = ({ children }: PropsWithChildren<{}>) => {
    const hasSearchContext = useSearchContextCheck()
    if(!hasSearchContext){
       return <>{children}</>;
    }
    
    return <SearchContextProvider>{children}</SearchContextProvider>;
}
```

**Review by emmaindal — COMMENTED** [review]
Just one small comment from my side, but otherwise it looks 💯

File: plugins/search/src/components/SearchModal/SearchModal.tsx (hunk 2 above)
Comment (camilaibs) [reply plugins/search/src/components/SearchModal/SearchModal.tsx:173]: Great idea, I will do it!

**Review by emmaindal — APPROVED** [review]
💯

**Pull request merged**